src/components/createContainer.jsx

```diff
@@ -0,0 +1,108 @@
+/* @flow */
+
+import React, { Component, PropTypes } from 'react';
+import invariant from 'invariant';
+
+import getDisplayName from '../utils/getDisplayName';
+import createAdaptorShape from '../utils/createAdaptorShape';
+
+const adaptorShape = createAdaptorShape(PropTypes);
+
+export default function createContainer(DecoratedComponent, specs) {
+  const displayName = `AdrenalineContainer(${getDisplayName(DecoratedComponent)})`;
+
+  invariant(
+    specs !== null && specs !== undefined,
+    `${displayName} requires configuration.`
+  );
+  invariant(
+    typeof specs.queries === 'function',
+    `You have to define 'queries' as a function.`
+  );
+
+  return class extends Component {
+    static displayName = displayName
+    static DecoratedComponent = DecoratedComponent
+
+    static contextTypes = {
+      renderLoading: PropTypes.func.isRequired,
+      adaptor: adaptorShape.isRequired,
+    }
+
+    constructor(props, context) {
+      super(props, context);
+
+      const { adaptor } = context;
+      //DecoratedComponent.prototype.shouldComponentUpdate = adaptor.shouldComponentUpdate;
```

Comment:
Yes, I just not quite sure about how we should apply this logic and on what level. Thats why its commented for now.